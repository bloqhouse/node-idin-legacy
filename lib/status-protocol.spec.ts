import { formatStatusProtocolXML } from './status-protocol'

const constantDate = '2017-10-09T20:08:32.110Z';

(Date as any) = class extends Date {
  constructor() {
    super(constantDate)
  }
}

describe('format status protocol', () => {
  it('it works correctly', () => {
    expect(formatStatusProtocolXML('XXX')).toMatchSnapshot()
  })
})
