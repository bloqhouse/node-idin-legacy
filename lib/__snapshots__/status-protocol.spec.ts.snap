// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status protocol it works correctly 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><AcquirerStatusReq version=\\"1.0.0\\" productID=\\"NL:BVN:BankID:1.0\\" xmlns=\\"http://www.betaalvereniging.nl/iDx/messages/Merchant-Acquirer/1.0.0\\" xmlns:ds=\\"http://www.w3.org/2000/09/xmldsig#\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\"><createDateTimestamp>2017-10-09T20:08:32.110Z</createDateTimestamp><Merchant><merchantID>0030001645</merchantID><subID>0</subID></Merchant><Transaction><transactionID>0000000000000000</transactionID></Transaction><Signature xmlns=\\"http://www.w3.org/2000/09/xmldsig#\\"><SignedInfo><CanonicalizationMethod Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/><SignatureMethod Algorithm=\\"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256\\"/><Reference URI=\\"\\"><Transforms><Transform Algorithm=\\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\\"/><Transform Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/></Transforms><DigestMethod Algorithm=\\"http://www.w3.org/2001/04/xmlenc#sha256\\"/><DigestValue>r1hLnXYHO7CxjHWdArlJqRWJzgKgQMdBcjs75bdekBY=</DigestValue></Reference></SignedInfo><SignatureValue>n9PpgkEgeagkBrd8YafWeQy/rBBK4ABnRBbnYd+ZC7M49Dvt89+eTCm+yimCJ6o6XaZU6ko6spLoBHe6+/wcsL6ckkVN9JK7L7amRB+jv9DoD7lZgvSjudqUZFVwGEpsK9E8/GquLZb73a440qqptngO8w0W/N/lCYZ0JzPhfx3trazh7N1jR3/XmKVqqv7nrjL0K2uKxn4SnXGYvEoWkr6oTv+1OcJ5i8gkJyP2ENgh7pqYU2WIOEZ7gFMtv6Tb8GJgpkR4Wmy+cNsurqkiEe2srrsS52Zk+l93H1cbeNmyFWxC5+FMllCqxsZKG10d2rljrUY8aQNUwarpOMDzUQ==</SignatureValue><KeyInfo><KeyName>8c6415e72e553d2a9bda7afc9fcc0a5ec5d82841</KeyName></KeyInfo></Signature></AcquirerStatusReq>"`;
